### ISSUE TITLE
CMake Documentation Issue

### ISSUE THREAD

**Issue by oke-aditya** [pytorch/pytorch#37766]
t isn't mentioned in PyTorch docs about the Set()
Also, the Cmake file isn't quite complete and needs to be updated. Maybe new users don't know about it.

set(CMAKE_PREFIX_PATH "libtorch/share/cmake/Torch")

here as well as here

The CMake file needs to be clear in the documentation. It is causing trouble for many people.
#12449 #21976 #806 #12449

It's not a bug. But a doc improvement issue.

**Comment by oke-aditya** [comment]
```
project(custom_ops)
set(CMAKE_PREFIX_PATH "libtorch/share/cmake/Torch")
find_package(Torch REQUIRED)

add_executable(example-app example-app.cpp)
target_link_libraries(example-app "${TORCH_LIBRARIES}")
set_property(TARGET example-app PROPERTY CXX_STANDARD 14)

if (MSVC)
    file(GLOB TORCH_DLLS "${TORCH_INSTALL_PREFIX}/lib/*.dll")
    add_custom_command(TARGET example-app
                 POST_BUILD
                 COMMAND ${CMAKE_COMMAND} -E copy_if_different
                 ${TORCH_DLLS}
                 $<TARGET_FILE_DIR:example-app>)
endif (MSVC)
```
This is the proper CMakeLists.txt file. Please correct it.

**Comment by ngimel** [comment]
Thank you for reporting, we will accept documentation PR for this.

**Comment by avigupta2612** [comment]
To correct this, I just need to change 
```
cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
project(dcgan)

find_package(Torch REQUIRED)

add_executable(dcgan dcgan.cpp)
target_link_libraries(dcgan "${TORCH_LIBRARIES}")
set_property(TARGET dcgan PROPERTY CXX_STANDARD 14)
```

to 
```
cmake_minimum_required(VERSION 3.0 FATAL_ERROR)
set(CMAKE_PREFIX_PATH "libtorch/share/cmake/Torch")
project(dcgan)

find_package(Torch REQUIRED)

add_executable(dcgan dcgan.cpp)
target_link_libraries(dcgan "${TORCH_LIBRARIES}")
set_property(TARGET dcgan PROPERTY CXX_STANDARD 14)
``` 
[here](https://pytorch.org/tutorials/advanced/cpp_frontend.html) ?

**Comment by oke-aditya** [comment]
Yes there as well as [here](https://pytorch.org/tutorials/advanced/cpp_export.html) and also [here](https://pytorch.org/cppdocs/installing.html).
I am unsure of Pytorch docs improvement etc. Hence I am unsure of raising PR here. :(

**Comment by DineshRajanT** [comment]
> ```
> project(custom_ops)
> set(CMAKE_PREFIX_PATH "libtorch/share/cmake/Torch")
> find_package(Torch REQUIRED)
> 
> add_executable(example-app example-app.cpp)
> target_link_libraries(example-app "${TORCH_LIBRARIES}")
> set_property(TARGET example-app PROPERTY CXX_STANDARD 14)
> 
> if (MSVC)
>     file(GLOB TORCH_DLLS "${TORCH_INSTALL_PREFIX}/lib/*.dll")
>     add_custom_command(TARGET example-app
>                  POST_BUILD
>                  COMMAND ${CMAKE_COMMAND} -E copy_if_different
>                  ${TORCH_DLLS}
>                  $<TARGET_FILE_DIR:example-app>)
> endif (MSVC)
> ```
> 
> This is the proper CMakeLists.txt file. Please correct it.

I tried this...and it didn't work for me....
